lib/stormpath-sdk/resource/collection.rb

```diff
@@ -51,6 +51,10 @@ def limit limit
     self
   end
 
+  def size
+    current_page.items.size
```

Comment:
I made the changes, can you check?

Also since we are already in the collection class I would remove the PaginatedIterator module and just move the iterate method to the collection class, this is the only place it is used and I don't see the point of this method being in a separate module.